extension/httpfs/httpfs.cpp

```diff
@@ -496,16 +571,20 @@ void HTTPFileHandle::Initialize(FileOpener *opener) {
 	}
 
 	if (res->headers.find("Content-Length") == res->headers.end() || res->headers["Content-Length"].empty()) {
-		// There was no content-length header, we can not do range requests here
-		throw IOException("Server did not send Content-Length header, can not read from this file.");
+		// There was no content-length header, we can not do range requests here, so we set the length to 0
+		length = 0;
+	} else {
+		try {
+			length = std::stoll(res->headers["Content-Length"]);
+		} catch (std::invalid_argument &e) {
+			throw IOException("Invalid Content-Length header received: %s", res->headers["Content-Length"]);
+		} catch (std::out_of_range &e) {
+			throw IOException("Invalid Content-Length header received: %s", res->headers["Content-Length"]);
+		}
 	}
-
-	try {
-		length = std::stoll(res->headers["Content-Length"]);
-	} catch (std::invalid_argument &e) {
-		throw IOException("Invalid Content-Length header received: %s", res->headers["Content-Length"]);
-	} catch (std::out_of_range &e) {
-		throw IOException("Invalid Content-Length header received: %s", res->headers["Content-Length"]);
+	if (length == 0) {
+		// Try to fully download the file first
+		range_read = false;
```

Comment:
I think we might want to do the download here, that way we have the size available after the OpenFile as you would expect from the filesystem api?

Reply:
Alright-o